internal/strategies/flexible_rollout.go

```diff
@@ -0,0 +1,79 @@
+package strategies
+
+import (
+	"math/rand"
+	"strconv"
+	"time"
+
+	"github.com/Unleash/unleash-client-go/v3/context"
+	"github.com/Unleash/unleash-client-go/v3/strategy"
+)
+
+type stickiness string
+
+const (
+	defaultStickiness   stickiness = "default"
+	userIDStickiness    stickiness = "userId"
+	sessionIDStickiness stickiness = "sessionId"
+	randomStickiness    stickiness = "random"
+)
+
+type flexibleRolloutStrategy struct {
+	random *rand.Rand
+}
+
+// NewFlexibleRolloutStrategy creates a new instance of the flexible rollout strategy.
+func NewFlexibleRolloutStrategy() *flexibleRolloutStrategy {
+	s := &flexibleRolloutStrategy{
+		random: rand.New(rand.NewSource(time.Now().Unix())),
```

Comment:
It would be better to use e.g. `time.Now().UnixNano() + int64(os.Getpid())` here (this is what ioutil.TempFile is using to seed it's RNG). Alternatively, crypto/rand can be used to get a proper random seed. With the code in the PR there is a risk that several instances of an application get the same seed if they are all restarted at approximately the same time (e.g., during a deploy of a new version).

Reply:
Why not just use math/rand? Already used in [gradual_rollout_random.go](https://github.com/Unleash/unleash-client-go/blob/master/internal/strategies/gradual_rollout_random.go#L36)

Reply:
Fair point, but same seed is not really an issue is it? The number is not used for anything cryptographic and `crypto/rand` is an order of magnitude slower. Right now, the implementation here is similar to Java:

```java
random = new Random();
```

And Node

```js
Math.floor(Math.random() * 100) + 1
```

Happy to change this if there is consensus, but I feel performance is more important than unique seed here?

Reply:
+1 for performance.

Reply:
Btw, allowing to the seed will enable unit testing of the implementation.

Reply:
No, the code isn't similar to the Java or Node code. The Java doc says
> This constructor sets the seed of the random number generator to a value very likely to be distinct from any other invocation of this constructor.

From what I understand of the JS documentation `Math.random()` returns a uniformly distributed number if the range [0, 1).

Unless I have missed something `NewFlexibleRolloutStrategy` is only called once, if so using crypto/rand won't be a problem. But, sure, if you're worried about performance use `time.Now().UnixNano() + int64(os.Getpid())` instead.